src/client/datascience/interactive-common/interactiveBase.ts

```diff
@@ -676,7 +678,30 @@ export abstract class InteractiveBase extends WebviewPanelHost<IInteractiveWindo
                     if (debugInfo.runByLine && debugInfo.hashFileName) {
                         await this.jupyterDebugger.startRunByLine(this._notebook, debugInfo.hashFileName);
                     } else if (!debugInfo.runByLine) {
-                        await this.jupyterDebugger.startDebugging(this._notebook);
+                        const promise = this._notebook.execute(
+                            `import os;os.environ["IPYKERNEL_CELL_NAME"] = '${file.replace(/\\/g, '\\\\')}'`,
```

Comment:
This is a harmless env variable for older Ipyhotn, hence i think we can just run this everytime.
My plan is to optimize all of this with heavy caching (cache version checks & depdnencies, etc rather than checking everytime https://github.com/microsoft/vscode-jupyter/issues/6694)
I'd rather do that after this PR goes, i.e. work on optimization seprately

Reply:
@rchiodo /cc

Reply:
Maybe this code coudl be in a separate function/file? I know the dupe of this in the interactive window will go away but there's quite a bit here that's making the enclosing function rather large.

Reply:
Moved most of the code, hence this enclosing function now has just one new line (moved all fo the other code out).